tss-esapi/src/error/return_code/base.rs

```diff
@@ -0,0 +1,260 @@
+// Copyright 2022 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use crate::{constants::return_code::BaseError, Error, Result};
+use std::convert::TryFrom;
+
+/// Enum representing the TSS base response code
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+pub enum BaseReturnCode {
```

Comment:
There is actually some reason to this madness. When I did this I thought that all values like the actual numbers of the errors should be kept in ``` constants```. And the actual rust ```Errors``` should be kept under ```errors```. The BasereturnCode in its entire form is actually used by the ```ResourceManager``` layer. So yes I think this is needed.

Reply:
My question is less about the existence of the type, actually, more about the implementation of it. It feels like we're duplicating that enum (unless some of the variants are not common?), instead of just wrapping it up.

Reply:
Ah I see what you mean. I will look into it. Would be nice to not duplicate things.

Reply:
Fixed